logger.go

```diff
@@ -0,0 +1,53 @@
+package gocron
+
+// Logger interface the custom logger should
+// conform to
+type Logger interface {
```

Comment:
this is real cool. have you thought about if it makes sense to have other standard log levels? warn, trace, e.g.

Reply:
Yes... But I was thinking, this interface needs to be as simple and generic as possible. 
But also, once we release this, adding any new method to this interface would not be backward compatible. So, what do you think? Should I add all the other levels?

```go
type Logger interface {
	Debugf(format string, args ...interface{})
	Infof(format string, args ...interface{})
	Printf(format string, args ...interface{})
	Warnf(format string, args ...interface{})
	Warningf(format string, args ...interface{})
	Errorf(format string, args ...interface{})
	Fatalf(format string, args ...interface{})
	Panicf(format string, args ...interface{})

	Debug(args ...interface{})
	Info(args ...interface{})
	Print(args ...interface{})
	Warn(args ...interface{})
	Warning(args ...interface{})
	Error(args ...interface{})
	Fatal(args ...interface{})
	Panic(args ...interface{})
}
```

Reply:
This is just an idea: from what I can see the current idea is the user to pass a log that satisfies this interface so we can use it internally. What if we, instead of accepting this logger interface, accepted an io.Writer to know where to write data to. This way we would break this dependency and it'd be pretty easy to test too. Let's stress this further if it makes sense.

Reply:
Interesting thought. But the user will not have any control over the format of the log. Some people follow a specific structure in their logs, right?

Reply:
Taking a step back for a moment - what are we going to be logging? Errors primarily i would assume.

What if we add a channel to our scheduler or the job itself that we return job run errors on for the caller to handle however they so choose. Instead of adding any logging in the package itself?

Reply:
If we stop to think about it, errors on schedule. Anything else can be caught up by closures from within the job function the client passes itself... So I'm starting to question actually the real need for this indeed. When (if) we move from our own scheduling logic to using a time.Ticker/After approach we won't even need to worry about this anymore. What do you think @arjunmahishi?

Reply:
Makes sense... There is no real need for this, agreed.

Reply:
So, are we going with the error channel approach?
Or does this need more thought?

Reply:
@Streppel @JohnRoesler Waiting to hear your thoughts

Reply:
@arjunmahishi sorry! Pretty busy last couple of days and github emails got lost in my inbox. I'd say to keep this branch open while we stress this idea a little bit further. We could try to identify within the code what could go wrong in places that users wouldn't be able to handle by themselves.

One example that comes to mind [is this one](https://github.com/go-co-op/gocron/blob/04be48f61037ca7e035c0f17f4547c3b0ba93a0e/scheduler.go#L254), which could happen if a job fails to run or to schedule. We are safe against scheduling failures as we do all the checks on the job creation. And the running part fails if the scheduler is supposed to lock the job on a distributed system without a locker implementation ([something we know that isn't working at the moment and needs to be fixed](https://github.com/go-co-op/gocron/issues/50), and after the fix we could fail to start a job under such conditions, or maybe consider a "locking job"  if it has locker instead of doing so with a flag). So, in the end, I believe this error isn't worth considering.

And actually, if we look at it... Scheduling and running jobs is basically everything this lib needs to do. Could you guys point to any other places that you think would need extra logging?

Reply:
(btw fellow comrades let me know if any of you want to get the linked issue, otherwise I'll take it)

Reply:
Currently, these are the functions whose errors we are not handling
```
scheduler.runAndReschedule
locker.Lock
locker.Unlock
callJobFuncWithParams
```

Reply:
Afterall, I don't think we should be handling logs internally. I believe callJob and runAndReschedule shouldn't need to return errors, and if something goes wrong with the distrubuted locker, users would probably want a little more control other than logging on how to handle the error; agree? @arjunmahishi

Reply:
Yes. The user should be able to handle the error.

Reply:
So, would the error channel that John suggested be useful here?

The scheduler could have a `func (s *sched) CatchErrors() chan` method to expose the error channel.

Reply:
I believe that'd be the way to go! Exposing though a Errors() method or something along those lines, making sure the program doesn't freeze if the buffer gets filled

Reply:
OK then! Will close this PR and work a new one :)